core/src/main/java/org/apache/accumulo/core/client/mapreduce/AbstractInputFormat.java

```diff
@@ -426,6 +442,7 @@ protected static ClientConfiguration getClientConfiguration(JobContext context)
      *          the table name for which the scanner is configured
      * @since 1.6.0
      */
+    @Deprecated
     protected abstract void setupIterators(TaskAttemptContext context, Scanner scanner, String tableName, RangeInputSplit split);
```

Comment:
I think we could have source compatibility with a switch from Scanner to ScannerBase, but I can't think of a way to handle AccumuloInputSplit instead of only RangeInputSplits. It should be straightforward to just call a new `setupIterator` method from the existing (now deprecated) method, no? Are you just trying to reduce the amount of code in here?